Adds documentation on manual span propogation (#153)
apple/swift-distributed-tracing: Sources/Tracing/Docs.docc/Guides/InstrumentYourLibrary.md

While this code is very simple for illustration purposes, and it may seem surprising why there are two separate places where we need to call into user-code separately, in practice such situations can happen when using asynchronous network or database libraries which offer their API in terms of callbacks. Always consider if and when to restore context such that it makes sense for the end user.

#### Manual propogation

> There are circumstances where `task-local` variables are interrupted during normal execution flow. One common instance is when using

Reviewer: ```suggestion
There are circumstances where [task-local variable](https://developer.apple.com/documentation/swift/tasklocal) propagation may be interrupted. One common instance is when using
```
